src/NServiceBus.Persistence.DynamoDB.AcceptanceTests/When_adding_operations_to_shared_session.cs

```diff
@@ -34,7 +35,36 @@ public async Task Should_commit_changes_with_session()
                 }
             });
 
-            Assert.AreEqual(items.Count, 3);
+            Assert.AreEqual(3, items.Count);
+        }
+
+        [Test]
+        public async Task Should_rollback_changes_with_session()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<EndpointAttachingTransactionOperations>(e => e
+                    .DoNotFailOnErrorMessages()
+                    .When(c => c.SendLocal(new TriggerMessage { FailHandler = true })))
+                .Done(c => c.MessageReceived)
```

Comment:
Isn't this test now always running into a timeout? Or is the test automagically aborted with the simulated exception? If not would it make sense to check c.FailedMessages.Any()?

Reply:
Never mind I missed that we have set it before we throw